### PR TITLE
Eliminate ViscousSponge cache, fix unit test

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR eliminates the ViscousSponge cache, and fixes the ViscousSponge unit test, which was not being exercised.

This is a followup to #3446.

### CODE REVIEW

File: src/solver/types.jl [hunk 1]
```diff
@@ -120,7 +120,9 @@ diffuse_momentum(::FriersonDiffusion{DM}) where {DM} = DM
 diffuse_momentum(::Nothing) = false
 
 abstract type AbstractSponge end
+Base.Broadcast.broadcastable(x::AbstractSponge) = tuple(x)
 Base.@kwdef struct ViscousSponge{FT} <: AbstractSponge
+    zmax::FT
```
Comment (szy21) [inline src/solver/types.jl:125]: I have the same question that @Sbozzolo has for the Rayleigh sponge: Is it possible to move this out of the struct?

Comment (charleskawczynski) [reply src/solver/types.jl:125]: Yes, I'll update this to do that

**Review by szy21 — APPROVED** [review]
Thanks!

**Comment by charleskawczynski** [comment]
`Spaces.z_max` wasn't added until ClimaCore 0.14.18, so I need to update the compat. I could add a check, but the alternatives would be to dig into internals, which I think is not a good idea, or use `maximum`, which is not really performant.

**Pull request merged**